numpyro/contrib/tfp/mcmc.py

```diff
@@ -0,0 +1,233 @@
+# Copyright Contributors to the Pyro project.
+# SPDX-License-Identifier: Apache-2.0
+
+from abc import ABCMeta
+from collections import namedtuple
+import inspect
+
+from jax import random, tree_map, vmap
+from jax.flatten_util import ravel_pytree
+import jax.numpy as jnp
+import tensorflow_probability.substrates.jax as tfp
+
+from numpyro.infer import init_to_uniform
+from numpyro.infer.mcmc import MCMCKernel
+from numpyro.infer.util import initialize_model
+from numpyro.util import identity
+
+
+TFPKernelState = namedtuple('TFPKernelState', ['z', 'kernel_results', 'rng_key'])
```

Comment:
Any idea how these tfp kernels works with CUDA? Can we set the jax device and would things automatically run on the GPU?

Reply:
Yes, I just test that we can switch device using `numpyro.set_platform` as usual